### PR TITLE
feat(errorhandling): error handling service account controller

### PR DESCRIPTION
**Pull request by AnuragNagpure**
## Description

Implement new error handling for the service controller

## Why

Implement for all POST and PUT endpoints the new error handling.

## Issue

N/A Ref: CPLP-3589

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by Phil91** [comment]
@AnuragNagpure the target branch was set to `main` , please don't do that. I've changed it to the `dev` branch

File: src/administration/Administration.Service/BusinessLogic/ServiceAccountBusinessLogic.cs [hunk 1]
```diff
@@ -30,6 +31,7 @@
 using Org.Eclipse.TractusX.Portal.Backend.Provisioning.Library.Enums;
 using Org.Eclipse.TractusX.Portal.Backend.Provisioning.Library.Models;
 using Org.Eclipse.TractusX.Portal.Backend.Provisioning.Library.Service;
+using System.Xml;
```
Comment (Phil91) [inline src/administration/Administration.Service/BusinessLogic/ServiceAccountBusinessLogic.cs:34]: can be removed

File: src/administration/Administration.Service/BusinessLogic/ServiceAccountBusinessLogic.cs [hunk 2]
```diff
@@ -59,28 +61,30 @@ public async Task<ServiceAccountDetails> CreateOwnCompanyServiceAccountAsync(Ser
...
-            throw new ConflictException($"bpn not set for company {companyId}");
+            throw ConflictException.Create(AdministrationServiceAccountErrors.SERVICE_BPN_NOT_SET_CONFLICT, new ErrorParameter[] { new("companyId", companyId.ToString()) });
         }
 
         var unassignable = serviceAccountCreationInfos.UserRoleIds.Except(result.TechnicalUserRoleIds);
         if (unassignable.Any())
         {
-            throw new ControllerArgumentException($"The roles {string.Join(",", unassignable)} are not assignable to a service account", "userRoleIds");
+            throw ControllerArgumentException.Create(AdministrationServiceAccountErrors.SERVICE_ROLES_NOT_ASSIGN_ARGUMENT, new ErrorParameter[] { new("unassignable", string.Join(",", unassignable)), new("userRoleIds", string.Join(",", result.TechnicalUserRoleIds)) });
+
```
Comment (Phil91) [inline src/administration/Administration.Service/BusinessLogic/ServiceAccountBusinessLogic.cs:87]: please remove empty line

File: src/administration/Administration.Service/BusinessLogic/ServiceAccountBusinessLogic.cs [hunk 3]
```diff
@@ -59,28 +61,30 @@ public async Task<ServiceAccountDetails> CreateOwnCompanyServiceAccountAsync(Ser
     {
         if (serviceAccountCreationInfos.IamClientAuthMethod != IamClientAuthMethod.SECRET)
         {
-            throw new ControllerArgumentException("other authenticationType values than SECRET are not supported yet", "authenticationType"); //TODO implement other authenticationTypes
+            throw ControllerArgumentException.Create(AdministrationServiceAccountErrors.SERVICE_AUTH_SECRET_ARGUMENT, new ErrorParameter[] { new("authenticationType", serviceAccountCreationInfos.IamClientAuthMethod.ToString()) });//TODO implement other authenticationTypes
+
```
Comment (Phil91) [inline src/administration/Administration.Service/BusinessLogic/ServiceAccountBusinessLogic.cs:65]: please remove empty line

File: src/administration/Administration.Service/BusinessLogic/ServiceAccountBusinessLogic.cs (hunk 2 above)
Comment (Phil91) [reply src/administration/Administration.Service/BusinessLogic/ServiceAccountBusinessLogic.cs:87]: In line 91 we call `_serviceAccountCreation.CreateServiceAccountAsync` please adjust the error handling within this method to the new logic as well

File: src/administration/Administration.Service/ErrorHandling/AdministrationServiceAccountErrorMessageContainer.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+using Org.Eclipse.TractusX.Portal.Backend.Framework.ErrorHandling.Library;
+using System.Collections.Immutable;
+
+namespace Org.Eclipse.TractusX.Portal.Backend.Administration.ErrorHandling;
+
+public class AdministrationServiceAccountErrorMessageContainer : IErrorMessageContainer
+{
+    private static readonly IReadOnlyDictionary<int, string> _messageContainer = new Dictionary<AdministrationServiceAccountErrors, string> {
+                { AdministrationServiceAccountErrors.SERVICE_AUTH_SECRET_ARGUMENT, "other authenticationType values than SECRET are not supported yet , {authenticationType}" },
+                { AdministrationServiceAccountErrors.SERVICE_NAME_EMPTY_ARGUMENT,"name must not be empty, {name}"},
```
Comment (Phil91) [inline src/administration/Administration.Service/ErrorHandling/AdministrationServiceAccountErrorMessageContainer.cs:29]: please add a whitespace between the parameters, same applies for the following entries

File: src/administration/Administration.Service/ErrorHandling/AdministrationServiceAccountErrorMessageContainer.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+using Org.Eclipse.TractusX.Portal.Backend.Framework.ErrorHandling.Library;
+using System.Collections.Immutable;
+
+namespace Org.Eclipse.TractusX.Portal.Backend.Administration.ErrorHandling;
```
Comment (Phil91) [inline src/administration/Administration.Service/ErrorHandling/AdministrationServiceAccountErrorMessageContainer.cs:23]: the namespace isn't correct. Please change to:

`Org.Eclipse.TractusX.Portal.Backend.Administration.Service.ErrorHandling

**Review by Phil91 — CHANGES_REQUESTED** [review]

File: src/administration/Administration.Service/BusinessLogic/ServiceAccountBusinessLogic.cs (hunk 1 above)
Comment (AnuragNagpure) [reply src/administration/Administration.Service/BusinessLogic/ServiceAccountBusinessLogic.cs:34]: removed

File: src/administration/Administration.Service/BusinessLogic/ServiceAccountBusinessLogic.cs (hunk 2 above)
Comment (AnuragNagpure) [reply src/administration/Administration.Service/BusinessLogic/ServiceAccountBusinessLogic.cs]: added required exception

File: src/administration/Administration.Service/BusinessLogic/ServiceAccountBusinessLogic.cs (hunk 3 above)
Comment (AnuragNagpure) [reply src/administration/Administration.Service/BusinessLogic/ServiceAccountBusinessLogic.cs]: removed

File: src/administration/Administration.Service/ErrorHandling/AdministrationServiceAccountErrorMessageContainer.cs (hunk 4 above)
Comment (AnuragNagpure) [reply src/administration/Administration.Service/ErrorHandling/AdministrationServiceAccountErrorMessageContainer.cs]: added

File: src/administration/Administration.Service/ErrorHandling/AdministrationServiceAccountErrorMessageContainer.cs (hunk 5 above)
Comment (AnuragNagpure) [reply src/administration/Administration.Service/ErrorHandling/AdministrationServiceAccountErrorMessageContainer.cs]: changed as per mentioned above.

**Review by Phil91 — APPROVED** [review]

**Pull request merged**